src/main/java/uk/gov/pay/directdebit/mandate/services/MandateService.java

```diff
@@ -78,35 +80,40 @@ public MandateService(DirectDebitConfig directDebitConfig,
     }
 
     public Mandate createMandate(CreateMandateRequest createRequest, String accountExternalId) {
-        return gatewayAccountDao.findByExternalId(accountExternalId)
-                .map(gatewayAccount -> {
-                    // TODO:
-                    // when we introduce GoCardless gateway accounts to work with create mandate,
-                    // then modify appropriate mandate reference values
-                    MandateBankStatementReference mandateReference = MandateBankStatementReference.valueOf(
-                            PaymentProvider.SANDBOX.equals(gatewayAccount.getPaymentProvider()) ?
-                                    RandomStringUtils.randomAlphanumeric(18) : "gocardless-default");
-
-                    Mandate mandate = aMandate()
-                            .withGatewayAccount(gatewayAccount)
-                            .withExternalId(MandateExternalId.valueOf(RandomIdGenerator.newId()))
-                            .withMandateReference(mandateReference)
-                            .withServiceReference(createRequest.getReference())
-                            .withState(MandateState.CREATED)
-                            .withReturnUrl(createRequest.getReturnUrl())
-                            .withCreatedDate(ZonedDateTime.now(ZoneOffset.UTC))
-                            .build();
-
-                    LOGGER.info("Creating mandate external id {}", mandate.getExternalId());
-                    Long id = mandateDao.insert(mandate);
-                    mandate.setId(id);
-                    mandateStateUpdateService.mandateCreatedFor(mandate);
-                    return mandate;
-                })
-                .orElseThrow(() -> {
-                    LOGGER.error("Gateway account with id {} not found", accountExternalId);
-                    return new GatewayAccountNotFoundException(accountExternalId);
-                });
+        Optional<GatewayAccount> potentialGatewayAccount = gatewayAccountDao.findByExternalId(accountExternalId);
+
+        if (potentialGatewayAccount.isPresent()) {
+            if (potentialGatewayAccount.get().getAccessToken().isEmpty()) {
+                LOGGER.error("Gateway account with id {} does not have access token", accountExternalId);
+                throw new UnlinkedGCMerchantAccountException(accountExternalId);
+            }
```

Comment:
I think this can be inside the `.map`.
Generally we try to avoid using `.isPresent` as there's usually a more functional option, like using `.map` and `.orElseThrow` as we do here.